parsec/parsec.c

```diff
@@ -1543,17 +1551,24 @@ parsec_hash_find_deps(const parsec_taskpool_t *tp,
     return &hd->dependency;
 }
 
-static int
-parsec_update_deps_with_counter(const parsec_taskpool_t *tp,
+int
+parsec_update_deps_with_counter(parsec_taskpool_t *tp,
                                 const parsec_task_t* restrict task,
-                                parsec_dependency_t *deps)
+                                parsec_dependency_t *deps,
+                                const parsec_task_t* restrict origin,
+                                const parsec_flow_t* restrict origin_flow,
+                                const parsec_flow_t* restrict dest_flow)
 {
     parsec_dependency_t dep_new_value, dep_cur_value;
 #if defined(PARSEC_DEBUG_PARANOID) || defined(PARSEC_DEBUG_NOISIER)
     char tmp[MAX_TASK_STRLEN];
     parsec_task_snprintf(tmp, MAX_TASK_STRLEN, task);
 #endif
 
+    (void)origin;
```

Comment:
Why do we add the `origin`, `origin_flow`, and `dest_flow` if we don't use it?

Reply:
Some termination detectors (e.g. Credit Distribution Detection) need to inject additional data inside the messages themselves, this part of the API allows this.

Reply:
Those types are not marked as resolved....

There are two possibilities here:
  - we keep this part of the API, to enable those termination detectors in the future, but it's untested functionality
  - we remove that part of the API, so it's cleaner, and re-enabling those detectors will be a much harder task